base/src/main/java/uk/ac/ebi/atlas/model/card/CardModelAdapter.java

```diff
@@ -0,0 +1,54 @@
+package uk.ac.ebi.atlas.model.card;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.List;
+import java.util.Optional;
+
+public class CardModelAdapter {
+
```

Comment:
Remember that any `JsonObject` can be backed by a `Map` and any `JsonArray` by a `List`. Gson will do the rest for you. I think you can simplify a bit of code here and rely less on side-effects (i.e. `forEach`).

Reply:
Having the `serialize` method just return a string would make it extremely difficult to test. And if it returns `Map<String, Object>`, then it shouldn't really be called serialize anymore. Or last option, it still returns a `JsonObject` but then the implementation becomes overly convoluted and you start mixing these `Maps` and `JsonObjects` everywhere and it all gets confusing...